lib/Cake/Model/Datasource/Database/Sqlite.php

```diff
@@ -63,13 +63,17 @@ class Sqlite extends DboSource {
 /**
  * SQLite3 column definition
  *
+ * @link https://www.sqlite.org/datatype3.html Datatypes In SQLite Version 3
+ *
  * @var array
  */
 	public $columns = array(
 		'primary_key' => array('name' => 'integer primary key autoincrement'),
 		'string' => array('name' => 'varchar', 'limit' => '255'),
 		'text' => array('name' => 'text'),
 		'integer' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'smallint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
+		'tinyint' => array('name' => 'integer', 'limit' => null, 'formatter' => 'intval'),
```

Comment:
OK I updated `ModelTask` and `FormHelper`. I'm not 100% about the rest. It looks like `DatatypeFixture` should be the right place to add `tiny_int` and `small_int` columns, since `biginteger` and `bool` are already tested there...